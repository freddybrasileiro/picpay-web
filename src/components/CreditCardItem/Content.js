import styled from "styled-components";

export default styled.section`
  min-width: 150px;
  display: flex;
  align-items: center;
`;
