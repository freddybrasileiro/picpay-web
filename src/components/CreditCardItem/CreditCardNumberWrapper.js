import styled from "styled-components";

export default styled.p`
  flex: 1;
  margin: 0 !important;
`;
