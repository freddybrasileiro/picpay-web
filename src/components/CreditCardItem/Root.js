import styled from "styled-components";

export default styled.section`
  display: flex;
  height: 36px;
  background-color: ${props =>
    props.theme.isSelected ? "rgba(125, 230, 198, 0.24)" : "transparent"};
  justify-content: center;
  width: 100%;
  cursor: pointer;

  :hover {
    background-color: rgba(125, 230, 198, 0.15);
  }

  div {
    color: ${props => (props.theme.width === "xs" ? "white" : "#474971")};
    flex: 1;
    margin-right: ${props => (props.theme.isSelected ? "12px" : "24px")};
    margin-left: 8px;
    font-weight: bold;
    display: flex;
  }
`;
