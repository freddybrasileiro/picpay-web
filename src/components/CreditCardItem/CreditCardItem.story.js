import React from "react";
import { storiesOf } from "@storybook/react";
import { ThemeProvider } from "styled-components";
import { IntlProvider } from "react-intl";

import styledComponentsTheme from "styledComponentsTheme";
import { DEFAULT_LOCALE, translationMessages } from "i18n";

import CreditCardItem from "components/CreditCardItem";

import card from "components/__fixtures__/card.fixture";

storiesOf("Components/CreditCardItem", module)
  .addDecorator(story => (
    <IntlProvider
      locale={DEFAULT_LOCALE}
      messages={translationMessages[DEFAULT_LOCALE]}
    >
      <ThemeProvider theme={styledComponentsTheme}>
        <div style={{ maxWidth: 500, margin: "10px 10px" }}>{story()}</div>
      </ThemeProvider>
    </IntlProvider>
  ))
  .add("Default", () => <CreditCardItem card={card} />)
  .add("Selected", () => <CreditCardItem card={card} isSelected />);
