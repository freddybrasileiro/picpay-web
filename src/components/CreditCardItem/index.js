import React from "react";
import PropTypes from "prop-types";
import { Typography, withWidth } from "@material-ui/core";

import Root from "./Root";
import Content from "./Content";
import CreditCard from "components/icons/CreditCard";
import Check from "components/icons/Check";
import CreditCardGreen from "components/icons/CreditCardGreen";

const Receipt = ({ width, card, isSelected }) => (
  <Root theme={{ isSelected }}>
    <Content>
      {width === "xs" ? (
        <CreditCardGreen width="14" height="14" />
      ) : (
        <CreditCard width="14" height="14" />
      )}
      <Typography variant="display2" component="p">
        {card.card_number}
      </Typography>
      {isSelected && <Check width="12" height="12" />}
    </Content>
  </Root>
);

const propTypes = {
  card: PropTypes.object,
  isSelected: PropTypes.bool
};

const defaultProps = {
  card: {},
  isSelected: false
};

Receipt.propTypes = propTypes;
Receipt.defaultProps = defaultProps;

export default withWidth()(Receipt);
