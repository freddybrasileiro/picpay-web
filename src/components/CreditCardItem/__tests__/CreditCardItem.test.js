import React from "react";
import { createShallow } from "@material-ui/core/test-utils";

import CreditCardItem from "components/CreditCardItem";

import card from "components/__fixtures__/card.fixture";

describe("CreditCardItem", () => {
  const shallow = createShallow({ dive: true });

  it("Should render the CreditCardItem", () => {
    expect(shallow(<CreditCardItem card={card} />)).toMatchSnapshot();
  });
});
