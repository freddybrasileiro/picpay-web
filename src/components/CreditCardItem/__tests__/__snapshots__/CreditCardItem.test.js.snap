// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreditCardItem Should render the CreditCardItem 1`] = `
<Fragment>
  <CreditCardItem
    card={
      Object {
        "brand": "Elo",
        "card_number": "1111111111111111",
        "cvv": "123",
        "expiry_date": "11/2021",
        "name": "Joao Silva",
        "zip": "04000000",
      }
    }
    isSelected={false}
    width="md"
  />
  <EventListener
    onResize={[Function]}
    target="window"
  />
</Fragment>
`;
